### PR TITLE
fix: add recaptcha form validation check

### LINKED ISSUE #6: Check form input fields "required" attribute and field's content before submitting form
Currently the recaptcha script does not check for any input fields that have the "required" attribute set. This skips the field validation process and submits the form regardless of any set input field values.

### PR DESCRIPTION
**Pull request by fmedv**
Uses [checkValidity()](https://caniuse.com/form-validation) function to implement simple [form validation check](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/checkValidity) to prevent form submission if form is invalid.

closes #6

### CODE REVIEW

**Comment by passioneight** [comment]
@fmedv thanks for the contribution! Very much appreciated.
I'll test it over the weekend and get back to you soon :-)

**Comment by fmedv** [comment]
@passioneight sure thing! It's only two lines of code I've added

Thank you for that awesome repo ;)

**Comment by passioneight** [comment]
@fmedv here's my two cents: 

I agree that we should consider the `required` attribute of any fields to avoid submitting invalid forms. However, using `checkValidity` will not display any hints to the user as the browser normally would:

image

So, I think we should go for the `reportValidity` instead. [Here's an example](https://codepen.io/chriscoyier/pen/mOzgBZ). Though, if we do, we also need to consider the [`novalidate`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form#attr-novalidate) attribute of the form itself. Because we'd hinder form submission if the validation was explicitly disabled (e.g., when using tools like [Parsley JS](https://parsleyjs.org/doc/examples/simple.html)).

What do you think? :-)

File: src/Resources/public/js/recaptcha.js [hunk 1]
```diff
@@ -33,17 +33,18 @@ googleRecaptchaScript.addEventListener('load', function() {
 
                             form.querySelector(_config.googleRecaptcha.querySelector).value = token;
 
-                            var doSubmit = form.dispatchEvent(new Event('submit', {
-                                cancelable: true
-                            }));
-
-                            if(doSubmit) {
-                                form.submit();
+                            if (form.checkValidity()) {
```
Comment (passioneight) [inline src/Resources/public/js/recaptcha.js]: `if (form.noValidate || form.reportValidity()) {`

**Review by passioneight — CHANGES_REQUESTED** [review]

**Comment by fmedv** [comment]
Sounds perfect. Thanks for your feedback!

**Review by fmedv — COMMENTED** [review]
Perfect

**Review by passioneight — APPROVED** [review]

**Comment by passioneight** [comment]
Thanks, @fmedv  :-)

**Pull request merged**